logstash-core/lib/logstash/pipeline.rb

```diff
@@ -15,681 +15,24 @@
 # specific language governing permissions and limitations
 # under the License.
 
-require "thread"
-require "stud/interval"
-require "concurrent"
-require "logstash-core/logstash-core"
-require "logstash/event"
-require "logstash/config/file"
-require "logstash/filters/base"
-require "logstash/inputs/base"
-require "logstash/outputs/base"
-require "logstash/instrument/collector"
-require "logstash/filter_delegator"
-require "logstash/compiler"
+# for backward compatibility
+# logstash-devutils-1.3.6 logstash_helpers has dependency on this class
+module LogStash
+  class Pipeline
```

Comment:
NOTE: any plugin doing `require 'logstash/pipeline'` (e.g. in specs) will be broken by this.
Wonder if it was considered to have this file around as:
```ruby
require 'logstash/java_pipeline'
module LogStash
  Pipeline = JavaPipeline
end
```

Reply:
This is a nice suggestion. I was thinking to remove everything completely including the plugins.